eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -0,0 +1,201 @@
+     */
+    private void checkTypeArguments(DetailAST typeToken) {
+        final DetailAST identToken = typeToken.findFirstToken(TokenTypes.IDENT);
+        // if there is no IDENT token, implies the token is default java types.
+        if (identToken != null) {
+            final String typeName = identToken.getText();
+            // if not exist in the classPathMap, that implies the type is java default types, such as int.
+            if (classPathMap.containsKey(typeName)) {
+                if (!isValidLibrary(classPathMap.get(typeName))) {
+                    log(typeToken, String.format(EXTERNAL_DEPENDENCY_ERROR, typeName));
+                } else {
+                    // Checks multiple type arguments
+                    final DetailAST typeArgumentsToken = typeToken.findFirstToken(TokenTypes.TYPE_ARGUMENTS);
+                    if (typeArgumentsToken != null) {
+                        for (DetailAST ast = typeArgumentsToken.getFirstChild(); ast != null; ast = ast.getNextSibling()) {
+                            if (ast.getType() == TokenTypes.TYPE_ARGUMENT) {
+                                checkSingleTypeArgument(ast);
+                            }
+                        }
+                    }
+                    // Checks single type argument
+                    final DetailAST typeArgumentToken = typeToken.findFirstToken(TokenTypes.TYPE_ARGUMENT);
+                    if (typeArgumentToken != null) {
+                        checkSingleTypeArgument(typeArgumentToken);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     *  A helper function that checks for single type arguments scenario.
+     *
+     * @param typeArgumentToken TYPE_ARGUMENT AST node
+     */
+    private void checkSingleTypeArgument(DetailAST typeArgumentToken) {
+        final DetailAST identToken = typeArgumentToken.findFirstToken(TokenTypes.IDENT);
+        // if there is no IDENT token, implies the token is default java types.
+        if (identToken != null) {
+            final String typeName = identToken.getText();
+            // if not exist in the classPathMap, that implies the type is java default types, such as int.
+            if (classPathMap.containsKey(typeName)) {
+                if (!isValidLibrary(classPathMap.get(typeName))) {
+                    log(typeArgumentToken, String.format(EXTERNAL_DEPENDENCY_ERROR, typeName));
+                }
+            }
+        }
+    }
+
+    /**
+     *  A helper function that checks for whether a class is from a valid internal dependency or is a suppression class
+     *
+     * @param classPath the full class path for a given class
+     * @return true if the class  is a suppression class, otherwise, return false.
+     */
+    private boolean isValidLibrary(String classPath) {
+        boolean isValidLibrary = false;
+        for (String lib : validLibrarySet) {
+            if (classPath.startsWith(lib)) {
+                isValidLibrary = true;
```

Comment:
You can `break;` the `for` loop if the library is already found, so it doesn't keep iterating through it.

Reply:
Thinking about this, it is easier and probably more readable to use `.stream()`
`return validLibrarySet.stream().anyMatch(x -> classPath.startsWith(lib));`